<a href="https://colab.research.google.com/github/emgrimm42/NLP-Hausarbeit/blob/main/ha_potenziellezeitungen.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Ziel dieses Pads ist es zu checken wie viele Volltext Seiten jeweilige potenzielle Zeitungsquellen für die HA haben.
Hier ist der Code basierend auf den drei Dresdener Versioen von "Der Freiheitskampf":

In [ ]:
# Python-Bibliothek pysolr installieren
%pip install -q pysolr

In [ ]:
import pysolr
import pandas as pd

# Solr-Endpunkt
SOLR_URL = "https://api.deutsche-digitale-bibliothek.de/2/search/index/newspaper-issues"

# Verbindung zum Solr-Index herstellen
solr = pysolr.Solr(SOLR_URL, always_commit=True, timeout=10)

# Suchquery
# zdb_id legt fest welche Zeitung und type ob die Metadaten(issue) oder den Volltext(page)
query = '(zdb_id:"2803953-1" AND type:"page") OR (zdb_id:"2803964-6" AND type:"page") OR (zdb_id:"2803974-9" AND type:"page")'

# Anzahl der Treffer abfragen (rows=0 reicht, wenn nur numFound benötigt wird)
results = solr.search(query, rows=1000)

# Treffer in DataFrame überführen
docs = results.docs
df = pd.DataFrame(docs)

# Gesamtzahl der Ausgaben
num_issues = int(results.hits)

print("Anzahl der digitalisierten Seiten:", num_issues)

1. Alternative: Deutsche Reichtagszeitung (1848) - 2794469-4

In [ ]:
import pysolr
import pandas as pd

# Solr-Endpunkt
SOLR_URL = "https://api.deutsche-digitale-bibliothek.de/2/search/index/newspaper-issues"

# Verbindung zum Solr-Index herstellen
solr = pysolr.Solr(SOLR_URL, always_commit=True, timeout=10)

# Suchquery
# zdb_id legt fest welche Zeitung und type ob die Metadaten(issue) oder den Volltext(page)
query = 'zdb_id:"2794469-4" AND type:"page"'

# Anzahl der Treffer abfragen (rows=0 reicht, wenn nur numFound benötigt wird)
results = solr.search(query, rows=1000)

# Treffer in DataFrame überführen
docs = results.docs
df = pd.DataFrame(docs)

# Gesamtzahl der Ausgaben
num_issues = int(results.hits)

print("Anzahl der digitalisierten Seiten:", num_issues)

Anzahl der digitalisierten Seiten: 494


2. Alternative Badener Lanzarett Teitung (1916-1918)

In [ ]:
import pysolr
import pandas as pd

# Solr-Endpunkt
SOLR_URL = "https://api.deutsche-digitale-bibliothek.de/2/search/index/newspaper-issues"

# Verbindung zum Solr-Index herstellen
solr = pysolr.Solr(SOLR_URL, always_commit=True, timeout=10)

# Suchquery
# zdb_id legt fest welche Zeitung und type ob die Metadaten(issue) oder den Volltext(page)
query = 'zdb_id:"2224323-9" AND type:"page"'

# Anzahl der Treffer abfragen (rows=0 reicht, wenn nur numFound benötigt wird)
results = solr.search(query, rows=1000)

# Treffer in DataFrame überführen
docs = results.docs
df = pd.DataFrame(docs)

# Gesamtzahl der Ausgaben
num_issues = int(results.hits)

print("Anzahl der digitalisierten Seiten:", num_issues)

Anzahl der digitalisierten Seiten: 348


3. Alternative: Deutsche Zeitung (1847-1850) - 2751569-2

In [ ]:
import pysolr
import pandas as pd

# Solr-Endpunkt
SOLR_URL = "https://api.deutsche-digitale-bibliothek.de/2/search/index/newspaper-issues"

# Verbindung zum Solr-Index herstellen
solr = pysolr.Solr(SOLR_URL, always_commit=True, timeout=10)

# Suchquery
# zdb_id legt fest welche Zeitung und type ob die Metadaten(issue) oder den Volltext(page)
query = 'zdb_id:"2751569-2" AND type:"page"'

# Anzahl der Treffer abfragen (rows=0 reicht, wenn nur numFound benötigt wird)
results = solr.search(query, rows=8000)

# Treffer in DataFrame überführen
docs = results.docs
df = pd.DataFrame(docs)

# Gesamtzahl der Ausgaben
num_issues = int(results.hits)

print("Anzahl der digitalisierten Seiten:", num_issues)

Anzahl der digitalisierten Seiten: 7666


Fazit: 7666 Seiten sind genug für eine Analyse (von Dr. Oberbichler abgesegnet). Nächster schritt ist also den Volltext in eine wahrlich monströse Excel Datei umzuwandeln.

In [ ]:
df.to_excel('Volltext_2751569-2.xlsx')